overlord/ifacestate/ifacemgr_test.go

```diff
@@ -179,3 +184,65 @@ func (s *interfaceManagerSuite) addPlugSlotAndInterface(c *C) {
 		Snap: &snap.Info{SuggestedName: "consumer"}, Name: "plug", Interface: "test"}})
 	c.Assert(err, IsNil)
 }
+
+func (s *interfaceManagerSuite) TestDoSetupSnapSecuirty(c *C) {
+	dirs.SetRootDir(c.MkDir())
+	defer dirs.SetRootDir("")
+
+	parserCmd := testutil.MockCommand(c, "apparmor_parser", "")
+	defer parserCmd.Restore()
+
+	osSnap := &snap.Info{
+		Type:          snap.TypeOS,
+		SuggestedName: "ubuntu-core",
+		Slots:         make(map[string]*snap.SlotInfo),
+	}
+	snap.AddImplicitSlots(osSnap)
+	err := s.mgr.Repository().AddSnap(osSnap)
+	c.Assert(err, IsNil)
+
+	dname := filepath.Join(dirs.SnapSnapsDir, "snap", "0", "meta")
+	fname := filepath.Join(dname, "snap.yaml")
+
+	data := []byte(`
+name: snap
+version: 1
+apps:
+ app:
+   command: foo
+plugs:
+ network:
+  interface: network
+`)
+	err = os.MkdirAll(dname, 0755)
+	c.Assert(err, IsNil)
+	err = ioutil.WriteFile(fname, data, 0644)
+	c.Assert(err, IsNil)
+
+	s.state.Lock()
+	task := s.state.NewTask("setup-snap-security", "")
+	ss := snapstate.SnapSetup{Name: "snap"}
+	task.Set("snap-setup-task", task.ID())
+	task.Set("snap-setup", ss)
+	taskset := state.NewTaskSet(task)
+	change := s.state.NewChange("test", "")
+	change.AddAll(taskset)
+	s.state.Unlock()
+
+	s.mgr.Ensure()
+	s.mgr.Wait()
+	s.mgr.Stop()
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	c.Check(task.Status(), Equals, state.DoneStatus)
+	c.Check(change.Status(), Equals, state.DoneStatus)
+
+	var conns map[string]ifacestate.ConnState
```

Comment:
Please unmarshal to a map rather than to ConnState, so you observe the bug reported above.